File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnMoodgridProvider.cs

namespace GracenoteSDK {

/**
** <b>Experimental</b>: GnMoodgridProvider
*/
public class GnMoodgridProvider : global::System.IDisposable {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnMoodgridProvider(global::System.IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnMoodgridProvider obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnMoodgridProvider() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnMoodgridProvider(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
    }
  }

/**
* Retrieves the name of the moodgrid provider.
* @return string representing the name of the provider.
*/
  public string Name {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnMoodgridProvider_Name_get(swigCPtr) );
	} 

  }

/**
* Retrieves the type of Moodgrid provider.e.g. playlist collection
* @return string value denoting type of provider
*/
  public string Type {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnMoodgridProvider_Type_get(swigCPtr) );
	} 

  }

/**
* Retrieves a bool value indicating whether the provider needs network access.
* @return requiresnetwork
*/
  public bool RequiresNetwork {
    get {
      bool ret = gnsdk_csharp_marshalPINVOKE.GnMoodgridProvider_RequiresNetwork_get(swigCPtr);
      return ret;
    } 
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnMoodgridProvider.java

package com.gracenote.gnsdk;

/** 
** <b>Experimental</b>: {@link GnMoodgridProvider} 
*/ 
 
public class GnMoodgridProvider {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnMoodgridProvider(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnMoodgridProvider obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnMoodgridProvider(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

/** 
* Retrieves the name of the moodgrid provider. 
* @return string representing the name of the provider. 
*/ 
 
  public String name() throws com.gracenote.gnsdk.GnException {
    return gnsdk_javaJNI.GnMoodgridProvider_name(swigCPtr, this);
  }

/** 
* Retrieves the type of Moodgrid provider.e.g. playlist collection 
* @return string value denoting type of provider 
*/ 
 
  public String type() throws com.gracenote.gnsdk.GnException {
    return gnsdk_javaJNI.GnMoodgridProvider_type(swigCPtr, this);
  }

/** 
* Retrieves a bool value indicating whether the provider needs network access. 
* @return requiresnetwork 
*/ 
 
  public boolean requiresNetwork() throws com.gracenote.gnsdk.GnException {
    return gnsdk_javaJNI.GnMoodgridProvider_requiresNetwork(swigCPtr, this);
  }

}
